### PR TITLE
Weight moderator rings by r instead of r^2

### PR DESCRIPTION
**Pull request by samuelshaner**
As was discussed in PR #268, this PR updated the ringify procedure to ringify cells that do not contain an outer bounding cylinder (a cylinder with a -1 halfspace) by r instead of r^2.

### CODE REVIEW

**Comment by wbinventor** [comment]
The documenation should be updated along with the code here, otherwise the changes look fine. I'm unsure why Travis failed here as well without swig. You could try changing `.travis.yml` to install swig directly by adding the following line under the `install` block:

```bash
  - sudo apt-get install swig
```

I just find this issue surprising since swig has been working just fine for our previous builds.

File: src/Cell.cpp [hunk 1]
```diff
@@ -1137,12 +1137,29 @@ void Cell::ringify(std::vector<Cell*>& subcells, double max_radius) {
...
   std::vector<Cell*>::iterator iter3;
 
-  /* Compute the area to fill with each equal volume ring */
-  double area = M_PI * fabs(radius1*radius1 - radius2*radius2) / _num_rings;
+  /* Compute the increment, either by radius or area, to use to construct
+   * the concentric rings */
+  double increment;
+
+  /* If there is no outer bounding surface, make the rings have the same
+  * radius increment. */
```
Comment (wbinventor) [inline src/Cell.cpp]: Might clarify this comment with something like "(e.g., moderator in a pin cell universes)"

File: src/Cell.cpp [hunk 2]
```diff
@@ -1137,12 +1137,29 @@ void Cell::ringify(std::vector<Cell*>& subcells, double max_radius) {
...
+  if (halfspace1 == 0)
+    increment = fabs(radius1 - radius2) / _num_rings;
+
+  /* If there is an outer bounding surface, make the rings have the same
```
Comment (wbinventor) [inline src/Cell.cpp]: Might clarify this comment with something like "(e.g., fuel in a pin cell universe)"

File: src/Cell.cpp [hunk 3]
```diff
@@ -1137,12 +1137,29 @@ void Cell::ringify(std::vector<Cell*>& subcells, double max_radius) {
...
 
   /* Generate successively smaller ZCylinders */
   for (int i=0; i < _num_rings-1; i++) {
-    radius2 = sqrt(radius1*radius1 - (area / M_PI));
+
+    /* Compute the outer radius of the next ring */
+    if (halfspace1 == 0)
+      radius2 = radius1 - increment;
+    else
+      radius2 = sqrt(radius1*radius1 - (increment / M_PI));
```
Comment (wbinventor) [inline src/Cell.cpp]: spaces around `*` operator (or no space around `/` operator)

**Comment by wbinventor** [comment]
Otherwise this should be a quick merge.

**Comment by samuelshaner** [comment]
The documentation has been updated and this PR has been merged with the PR #274, so that one should be merged first.

File: docs/source/usersguide/input.rst [hunk 4]
```diff
@@ -376,14 +376,27 @@ Rings and Sectors
...
 
-.. math::
-  R_\textit{eff} = \sqrt{\frac{\Delta x \Delta y}{\pi}}
+1. A ``Cell`` contains one ``ZCylinder`` with a negative halfspace.
+2. A ``Cell`` contains one ``ZCylinder`` with a negative halfspace and one ``ZCylinder`` with a positive halfspace.
+3. A ``Cell`` contains one ``ZCylinder`` with a positive halfspace.
+
+Rings for cases 1 and 2 are create such that each ring is of **equal volume**. Rings for cases 3 are created with equal **ring spacing**, where :math:`(R_{outer} - R_{inner})` is the same for all rings. The outer bounding ring for case 3 is set to the distance from the center of the universe that the cell is in to the corner of the parent lattice cell or, if one doesn't exist, the geometry bounding box. The rings for case 3 are chosen to have **equal spacing** instead of **equal volume** so that the inner ring (often a moderator ring next to a fuel pin) has a relatively small radius in order to capture the sharp flux gradient outside a fuel pin. The figure below shows a plot for the materials and plots of the cells for cases 1, 2, and 3 where 3 rings have been created for the corresponding cell in each case:
```
Comment (wbinventor) [inline docs/source/usersguide/input.rst]: typo: "are create" -> "are created"

**Pull request merged**